core/src/test/java/org/elasticsearch/test/ESBackcompatTestCase.java

```diff
@@ -141,6 +143,18 @@ private static Path backwardsCompatibilityPath() {
         return builder;
     }
 
+    @Override
+    public Settings indexSettings() {
+        // IndexMetaData.SETTING_DATA_PATH looks like it breaks backwards compatibility tests
+        Settings settings = super.indexSettings();
+        if (settings.get(IndexMetaData.SETTING_DATA_PATH) == null) {
+            return settings;
+        }
+        Settings.Builder builder = Settings.settingsBuilder().put(settings);
+        builder.remove(IndexMetaData.SETTING_DATA_PATH);
```

Comment:
I think you should log something here to make it apparent to anyone reading that the setting was removed? Maybe at `warn` level?